File: src/main/java/com/legacy/app/DatabaseConnection.java
package com.legacy.app;

public class DatabaseConnection {
    private String dbUrl = "jdbc:mysql://localhost:3306/warehouse";
    private String dbUser = "admin";
    
    // SECURITY VULNERABILITY: Hardcoded password
    // FIXED: Load password from environment variable
    private String dbPassword = System.getenv("DB_PASSWORD"); 

    public void connect() {
        if (dbPassword == null) {
            System.err.println("Error: DB_PASSWORD environment variable not set.");
            return;
        }

        System.out.println("Connecting to database...");
        System.out.println("Using user: " + dbUser);
        // FIXED: Do NOT print password to console
        System.out.println("Connected successfully (Mock)!");
    }
}

File: nodejs/src/databaseConnection.js
class DatabaseConnection {
    constructor() {
        this.dbUrl = "jdbc:mysql://localhost:3306/warehouse";
        this.dbUser = "admin";

        // SECURITY VULNERABILITY: Hardcoded password
        this.dbPassword = "admin123";
    }

    connect() {
        console.log("Connecting to database...");
        console.log("Using user: " + this.dbUser);
        // SECURITY VULNERABILITY: Printing password to console
        console.log("Using password: " + this.dbPassword);
        console.log("Connected successfully (Mock)!");
    }
}

module.exports = DatabaseConnection;
